playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/ConnectedBillingWrapper.java

```diff
@@ -0,0 +1,93 @@
+package com.google.androidbrowserhelper.playbilling.digitalgoods;
+
+import android.app.Activity;
+
+import com.android.billingclient.api.AcknowledgePurchaseResponseListener;
+import com.android.billingclient.api.BillingClientStateListener;
+import com.android.billingclient.api.BillingResult;
+import com.android.billingclient.api.ConsumeResponseListener;
+import com.android.billingclient.api.SkuDetails;
+import com.android.billingclient.api.SkuDetailsResponseListener;
+import com.google.androidbrowserhelper.playbilling.provider.BillingWrapper;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A wrapper around {@link BillingWrapper} that ensures it is connected before calling
+ * {@link #querySkuDetails}, {@link #acknowledge} or {@link #consume}.
+ */
+public class ConnectedBillingWrapper implements BillingWrapper {
+    private final BillingWrapper mInner;
+
+    private static final int NOT_CONNECTED = 0;
+    private static final int CONNECTING = 1;
+    private static final int CONNECTED = 2;
+    private int mState = NOT_CONNECTED;
+
+    private interface Callback {
```

Comment:
This is more of an observation: this is the same as the `Runnable` interface. Would it make sense to use it instead (or extend if we want to add extra methods)?

Reply:
Oh good spot, thanks!